server/src/main/scala/viper/gobraserver/GobraServer.scala

```diff
@@ -269,8 +272,8 @@ object GobraServer extends GobraFrontend {
   }
 
 
-  def stop(): Unit = {
-    _server.stop()
+  def stop(): Future[Unit] = {
+    _server.stop().map(_ => {})(_executor)
```

Comment:
I think in scala you usually use `()` as the unit literal. Here you are esentially creating a code block. This is probably optimized, but I would still go with `()`.